### ISSUE TITLE
Reset animation not working with more than one controller

### ISSUE THREAD

**Issue by erperejildo** [Klerith/animate_do_package#29]
I have an animation that I reset/stop ok when it's only rendered once. The problem comes when I have more instances of this controller, all of them created in a List, then, it restarts all the time.

```
  List<dynamic> blackBGControllerList = [];

for (var i = 0; i < players; i++) {
      var blackBGController;
      blackBGControllerList.add(blackBGController);
    }

void stopTimer(int index) async {
    if (timerControllerList[index].isAnimating) {
      blackBGControllerList[index].reset();
    }
}
```

App (don't mind the flashses, that's the emulator):
https://youtube.com/shorts/TlPtGNkyQ5U

Am I doing something wrong?

**Comment by erperejildo** [comment]
It seems that the problem comes with this setState here which is not related to this package:

```
for (var i = 0; i < players; i++) {
      var blackBGController;
      var finalMsgController;
      AnimationController timerController = AnimationController(
          duration: const Duration(seconds: 10), vsync: this);
      timerAnimation = Tween<double>(begin: 0, end: 10).animate(timerController)
        ..addListener(() {
          setState(() {}); // <--- HERE
        });
      blackBGControllerList.add(blackBGController);
      finalMsgControllerList.add(finalMsgController);
      timerControllerList.add(timerController);
    }
```

**Comment by erperejildo** [comment]
https://stackoverflow.com/questions/77455196/flutter-setstate-is-restarting-animation?noredirect=1#comment136550296_77455196

**Comment by erperejildo** [comment]
@Klerith any update on this? This is a small scenario. Remove the setState() to see it working:

```
import 'package:animate_do/animate_do.dart';
import 'package:bedtime/classes/loading.dart';
import 'package:bedtime/validators/validators.dart';
import 'package:flutter/material.dart';
import 'package:unicons/unicons.dart';

class NewStoryPage extends StatefulWidget {
  const NewStoryPage({super.key});
  @override
  NewStoryPageState createState() => NewStoryPageState();
}

class NewStoryPageState extends State<NewStoryPage> {
  final _nameFormKey = GlobalKey<FormState>();
  final _ageFormKey = GlobalKey<FormState>();
  final TextEditingController _nameController = TextEditingController();
  final TextEditingController _ageController = TextEditingController();
  late AnimationController nameAnimationController;
  late AnimationController ageAnimationController;
  int animationDuration = 2;
  int formStep = 0;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('New Story'),
      ),
      body: Padding(
        padding: const EdgeInsets.all(16),
        child: forms(),
      ),
      floatingActionButton: Row(
        mainAxisAlignment: MainAxisAlignment.end,
        children: [
          Visibility(
            visible: formStep > 0,
            child: backButton(),
          ),
          SizedBox(width: 16),
          nextButton(),
        ],
      ),
    );
  }

  Widget forms() {
    return Column(
      mainAxisAlignment: MainAxisAlignment.center,
      children: [
        Form(
          key: _nameFormKey,
          child: nameTextField(),
        ),
        Form(
          key: _ageFormKey,
          child: ageTextField(),
        ),
      ],
    );
  }

  Widget nameTextField() {
    return FadeIn(
      controller: (controller) => nameAnimationController = controller,
      duration: Duration(seconds: animationDuration),
      child: TextFormField(
        controller: _nameController,
        validator: (value) {
          return Validators.checkInput(value!);
        },
        // readOnly:
        //     Validators.checkInput(_nameController.text) == null ? true : false,
        decoration: InputDecoration(
          labelText: "What's the protagonist name?",
          floatingLabelBehavior: FloatingLabelBehavior.auto,
        ),
      ),
    );
  }

  Widget ageTextField() {
    return FadeIn(
      controller: (controller) => ageAnimationController = controller,
      manualTrigger: true,
      duration: Duration(seconds: animationDuration),
      child: TextFormField(
        controller: _ageController,
        validator: (value) {
          return Validators.checkInput(value!);
        },
        decoration: InputDecoration(
          labelText: "And how old is he/she?",
          floatingLabelBehavior: FloatingLabelBehavior.auto,
        ),
      ),
    );
  }

  Widget backButton() {
    return FloatingActionButton.extended(
      heroTag: null,
      onPressed: back,
      label: Text('Back'),
    );
  }

  Widget nextButton() {
    return FloatingActionButton.extended(
      heroTag: null,
      onPressed: next,
      label: Text('Next'),
    );
  }

  Future<void> back() async {}

  Future<void> next() async {
    if (!_nameFormKey.currentState!.validate()) return;

    if (formStep == 0) {
      await nameAnimationController.reverse();
      await ageAnimationController.forward();

      setState(() {
        formStep = 1;
      });
      return;
    }

    if (!_ageFormKey.currentState!.validate()) return;

    if (formStep == 1) {
      await ageAnimationController.reverse();
      // TODO
    }
  }
}

```

**Comment by Klerith** [comment]
@erperejildo I'm checking it right now, thank you for the example.

Also, I think for what you need to implement, the solution #30 will be what you need in here.

**Comment by Klerith** [comment]
In the meantime, @erperejildo, you can use the **animate** property; it will be easier than managing the controllers.

Check this out: I removed the animation controller and used boolean values instead.

```
import 'package:animate_do/animate_do.dart';
import 'package:flutter/material.dart';

class NewStoryPage extends StatefulWidget {
  const NewStoryPage({super.key});
  @override
  NewStoryPageState createState() => NewStoryPageState();
}

class NewStoryPageState extends State<NewStoryPage> {
  final _nameFormKey = GlobalKey<FormState>();
  final _ageFormKey = GlobalKey<FormState>();
  final TextEditingController _nameController = TextEditingController();
  final TextEditingController _ageController = TextEditingController();

  bool animateName = true;
  bool animateAge = false;

  
  int animationDuration = 2;
  int formStep = 0;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('New Story'),
      ),
      body: Padding(
        padding: const EdgeInsets.all(16),
        child: forms(),
      ),
      floatingActionButton: Row(
        mainAxisAlignment: MainAxisAlignment.end,
        children: [
          Visibility(
            visible: formStep > 0,
            child: backButton(),
          ),
          const SizedBox(width: 16),
          nextButton(),
        ],
      ),
    );
  }

  Widget forms() {
    return Column(
      mainAxisAlignment: MainAxisAlignment.center,
      children: [
        Form(
          key: _nameFormKey,
          child: nameTextField(),
        ),
        Form(
          key: _ageFormKey,
          child: ageTextField(),
        ),
      ],
    );
  }

  Widget nameTextField() {
    return FadeIn(
      animate: animateName,
      duration: Duration(seconds: animationDuration),
      child: TextFormField(
        decoration: const InputDecoration(
          labelText: "What's the protagonist name?",
          floatingLabelBehavior: FloatingLabelBehavior.auto,
        ),
      ),
    );
  }

  Widget ageTextField() {
    return FadeIn(
      animate: animateAge,
      duration: Duration(seconds: animationDuration),
      child: TextFormField(
        decoration: const InputDecoration(
          labelText: "And how old is he/she?",
          floatingLabelBehavior: FloatingLabelBehavior.auto,
        ),
      ),
    );
  }

  Widget backButton() {
    return FloatingActionButton.extended(
      heroTag: null,
      onPressed: back,
      label: const Text('Back'),
    );
  }

  Widget nextButton() {
    return FloatingActionButton.extended(
      heroTag: null,
      onPressed: next,
      label: const Text('Next'),
    );
  }

  Future<void> back() async {}

  Future<void> next() async {
    if (!_nameFormKey.currentState!.validate()) return;

    if (formStep == 0) {

      animateName = false;
      animateAge = true;

      setState(() {
        formStep = 1;
      });
      return;
    }

    if (!_ageFormKey.currentState!.validate()) return;

    if (formStep == 1) {
      animateName = true;
      animateAge = false;
    }
  }
}

```

**Comment by erperejildo** [comment]
> @erperejildo I'm checking it right now, thank you for the example.
> 
> Also, I think for what you need to implement, the solution #30 will be what you need in here.

as mentioned there, I can't use `onFinish`.

Don't worry, there is no rush. Since I needed this I ended up using something else so whenever it's ready and I have to implement new animations I'll use it

**Comment by Klerith** [comment]
Published v V3.2.0!

Now, I have worked on this issue because you can do it without the controller. However, I refactored some code that might have fixed that problem.

Feel free to reopen or create a new issue

**Comment by erperejildo** [comment]
gracias!
